spark/src/main/scala/ai/chronon/spark/JoinBase.scala

```diff
@@ -418,21 +418,23 @@ abstract class JoinBase(joinConf: api.Join,
              s"groupBy.metaData.team needs to be set for joinPart ${jp.groupBy.metaData.name}")
     }
 
-    // Run validations before starting the job
-    val today = tableUtils.partitionSpec.at(System.currentTimeMillis())
-    val analyzer = new Analyzer(tableUtils, joinConf, today, today, silenceMode = true)
-    try {
-      analyzer.analyzeJoin(joinConf, validationAssert = true)
-      metrics.gauge(Metrics.Name.validationSuccess, 1)
-      logger.info("Join conf validation succeeded. No error found.")
-    } catch {
-      case ex: AssertionError =>
-        metrics.gauge(Metrics.Name.validationFailure, 1)
-        logger.error(s"Validation failed. Please check the validation error in log.")
-        if (tableUtils.backfillValidationEnforced) throw ex
-      case e: Throwable =>
-        metrics.gauge(Metrics.Name.validationFailure, 1)
-        logger.error(s"An unexpected error occurred during validation. ${e.getMessage}")
+    // Run validations before starting the job; Skip validation if selectedJoinParts is defined
+    if (selectedJoinParts.isEmpty) {
```

Comment:
I am a little lost here. Should it run analyzer when selected join parts is not empty? 
```suggestion
    if (!selectedJoinParts.isEmpty) {
```

Reply:
When `selectedJoinParts` is not empty, it is a join part task. We don't have to run the analyzer on the join parts as they have been validated in the left table task and will be checked again in the final table task. Hence, we only run the analyzer for non JP use cases - when `selectedJoinParts` is empty.

Reply:
if the selected join parts are empty, do we still need to run analyzer? would it be an invalid join?

Reply:
Yes - we do. It is not only for backward compatibility, but also there are valid use cases like [here](https://github.com/airbnb/chronon/blob/97b06ec6c13b2289d86130b1bbb37de9d0aab998/spark/src/main/scala/ai/chronon/spark/stats/ConsistencyJob.scala#L84) without any selected join parts. That's why we didn't wipe out the analyzer from here.